### ISSUE TITLE
Dropdown list jump

### ISSUE THREAD

**Issue by anilastrio** [select2/select2#2940]
I notice when you click on the option, the dropdown list centers to that selected option... which is great. But when you select the second value and if the second value is lower in the list (below the first value), the list jumps back (centers) to the 1st value upon selecting the 2nd value. But if the value is higher in the list (above the first value), than it works fine. Its centers to the 2nd value.

Hope this makes sense.

**Comment by kevin-brown** [comment]
The dropdown  _should_ close automatically when a selection is made. So I'm not sure where the jumping comes into play, except for when you open the dropdown again.

When the dropdown is opened, the first value that is selected is focused, which is different from the old behaviour where it would always focus on the first item in the list.

**Comment by anilastrio** [comment]
I understand now. I require it to "keep open" so i added:
```javascript
 .on("select2:select", function () { 
						if ($(this).val().length > 4) {
							$(this).select2("close"); 
						} else {
							$(this).select2("open"); 
						}
					});
```
which results to the jumping.
Is there a solution whereby it can focus on the last selected item?

**Comment by kevin-brown** [comment]
After quite a bit of testing, I've determined that it's not possible for us to handle this edge case. We can't determine if an element is being hovered unless the user moves their mouse over it, and they would have to move it in a very short period of time after the list closes in order for us to handle it.

You _might_ be able to handle this by preventing the `select2:closing` event instead. For now I've marked this as a future enhancement rather than a current bug.
